pvanalytics/quality/irradiance.py

```diff
@@ -471,3 +471,174 @@ def daily_insolation_limits(irrad, clearsky, daily_min=0.4, daily_max=1.25):
+    elif dni is not None and ghi is not None and dhi is None:
+        dhi = (ghi - dni * cosd(solar_zenith))
+    else:
+        raise ValueError(
+            "Please check that exactly one of ghi, dhi and dni parameters "
+            "is set to None"
+        )
+    # Merge the outputs into a master dataframe containing 'ghi', 'dhi',
+    # and 'dni' columns
+    component_sum_df = pd.DataFrame({'ghi': ghi,
+                                     'dhi': dhi,
+                                     'dni': dni})
+    return component_sum_df
+
+
+def calculate_component_sum_series(solar_zenith,
+                                   ghi=None,
+                                   dhi=None,
+                                   dni=None,
+                                   dni_clear=None,
+                                   zenith_limit=90,
+                                   fill_value=np.nan,
+                                   fill_nighttime=None):
+    '''
+    Use the component sum equations to calculate the missing series, using
+    the other available time series. One of the three parameters (ghi, dhi,
+    dni) is passed as None, and the other associated series passed are used to
+    calculate the missing series value. After calculation, the series is
+    run through a nighttime routine, where nighttime values are set based on
+    the fill_value and fill_nighttime parameters.
+
+    The "component sum" or "closure" equation relates the three
+    primary irradiance components as follows:
+    .. math::
+       GHI = DHI + DNI * cos(theta_z)
+
+    Parameters
+    ----------
+    solar_zenith : Series
+        Zenith angles in decimal degrees, with datetime index.
+        Angles must be >=0 and <=180. Must have the same datetime index
+        as ghi, dhi, and dni series, when available.
+    ghi : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as dni, dhi, and zenith series, when available.
+    dhi : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as ghi, dni, and zenith series, when available.
+    dni : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as ghi, dhi, and zenith series, when available.
+    dni_clear : Series, optional
+        Pandas series of clearsky dni data. Must have the same datetime index
+        as ghi, dhi, dni, and zenith series, when available. See
+        :py:func:`dni` for details.
+    zenith_limit: Float
+        Solar zenith boundary between night and day, in degrees.
+        Zenith values greater than the limit are filled with a
+        constant default of 90.
+    fill_value: Float, default np.nan
+        The value that is used to fill in nighttime value
+    fill_nighttime: String, default None
+        Options include 'fill_value', 'equation', None. This is the
```

Comment:
This combination `fill_value` and `fill_nighttime` seems a bit redundant. `fill_nighttime='equation'` is equivalent to filling with DHI or GHI, so this behavior could be set by `fill_nighttime='GHI'` (or ='DHI'). `fill_nighttime='fill_value'` could be indicated if `fill_nighttime` is set to a numeric.

Reply:
@cwhanse I took a stab at addressing this in https://github.com/pvlib/pvanalytics/pull/163/commits/37eb655b50d66c9ca4cbbecc2cb339bc4be79bc0. Now we only have one variable fill_night_value, which can either be a float or int or 'equation'. Normally I don't like to do mixed types as a variable, but this does reduce the redundancy. Let me know what you think